### PR TITLE
Add `micronaut-aop` feature.

### LINKED ISSUE #1633: Add `micronaut-aop` feature
At the moment the only way to find the name of the `io.micronaut:micronaut-aop` dependency and the scope it should be used in (`compileOnly` for gradle) is reading a [blogpost(micronaut.io)](https://micronaut.io/2019/10/07/micronaut-aop-awesome-flexibility-without-the-complexity/) - no documentation. I think adding it to launch is a great way to show people this amazing feature. An alternative would be to add it to  add it to the almighty micronaut documentation page : https://docs.micronaut.io/latest/guide/#aop

### PR DESCRIPTION
**Pull request by wetted**
closes #1633

### CODE REVIEW

File: starter-core/src/main/java/io/micronaut/starter/feature/aop/AOP.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import io.micronaut.starter.build.dependencies.Priority;
+import io.micronaut.starter.feature.Category;
+import io.micronaut.starter.feature.Feature;
+import jakarta.inject.Singleton;
+
+@Singleton
+public class AOP implements Feature {
+    public static final String NAME = "micronaut-aop";
+
+    private static final Dependency MICRONAUT_INJECT_JAVA_DEPENDENCY = MicronautDependencyUtils
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/aop/AOP.java]: is this necessary?

File: starter-core/src/main/java/io/micronaut/starter/feature/aop/AOP.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    @Override
+    public String getCategory() {
+        return Category.API;
+    }
+
+    @Override
+    public void apply(GeneratorContext generatorContext) {
+        generatorContext.addDependency(MICRONAUT_INJECT_JAVA_DEPENDENCY);
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/aop/AOP.java]: is this necessary?

File: starter-core/src/main/java/io/micronaut/starter/feature/aop/AOP.java (hunk 1 above)
Comment (wetted) [reply starter-core/src/main/java/io/micronaut/starter/feature/aop/AOP.java]: @sdelamo I am going by instructions in @graemerocher blog post. Has that changed?

https://micronaut.io/2019/10/07/micronaut-aop-awesome-flexibility-without-the-complexity/

Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/aop/AOP.java]: our build plugins add the mironaut-inject-java annotation processor automatically for users.

**Review by sdelamo — APPROVED** [review]

**Pull request merged**